lib/_http_agent.js

```diff
@@ -240,6 +244,10 @@ function calculateServerName(options, req) {
   let servername = options.host;
   const hostHeader = req.getHeader('host');
   if (hostHeader) {
+    if (typeof hostHeader !== 'string') {
+      throw new ERR_HTTP_INVALID_HEADER_VALUE(hostHeader, 'host');
```

Comment:
`ERR_INVALID_ARG_TYPE` is likely the better choice here
```suggestion
      throw new ERR_INVALID_ARG_TYPE('options.headers.host', 'String', hostHeader);
```

Reply:
didn't see that error before, I applied the feedback